syntax = "proto3";

package connection;

service ConnectionService {
  rpc FindContacts(ConnectionRequest) returns (ConnectionResponse) {}
}

message ConnectionRequest {
  int32 person_id = 1;
  string start_date = 2;
  string end_date = 3;
  int32 meters = 4;
}

message ConnectionResponse {
  repeated Connection connections = 1;
}

message Connection {
  Location location = 1;
  Person person = 2;
}

message Location {
  int32 id = 1;
  int32 person_id = 2;
  string longitude = 3;
  string latitude = 4;
  string creation_time = 5;
}

message Person {
  int32 id = 1;
  string last_name = 2;
  string first_name = 3;
  string company_name = 4;
}
